include/drivers/irq/riscv_plic0.h

```diff
@@ -3,10 +3,18 @@
  * Copyright 2021, HENSOLDT Cyber
  *
  * SPDX-License-Identifier: GPL-2.0-only
+ *
+ * SiFive U54/U74 PLIC handling (HiFive Unleashed/Unmatched, Polarfire)
  */
 
 #pragma once
 
+/* This is a check that prevents using this driver blindly. Extend the list if
+ * this driver is confirmed to be working on other platforms. */
+#if !defined(CONFIG_PLAT_HIFIVE) && !defined(CONFIG_PLAT_POLARFIRE)
+#error "This code supports the SiFive U54/U74 PLIC only."
```

Comment:
This change seems premature to me based on the following assumptions:
- RISC-V is trying to have a standardized PLIC that is used across many implementations
- This driver is written for a device with a compatibility string `riscv,plic0` in the device tree, which doesn't indicate that it's SiFive specific as compatibility strings are supposed to be in a global namespace.
- We don't have any platforms in the kernel that need a different PLIC driver yet (ignoring the platforms that don't use a PLIC) and as far as we know, the next platforms will also need to use this driver.

Reply:
I've put this in as a marker. It is expected to be lifted again, but this should be a deliberate action then to track this better. You have a point that it is premature, but since RISC-V is a quite open platform where any chip could be special, I'd prefer to be restrictive here for a while and see what happens.

Reply:
But adding this header file to other platforms already needs to be a deliberate action in declare_default_headers() so this seems overly conservative to me.

Reply:
True, but we had some discussions internally about the PLIC spec, what our chip does, what other implementations seem to be doing and where the spec is evolving. There is a certain fear of fragmentation here and due to the lack of actual silicon form other vendors I tend to be very cautious here. I'm not too happy with `riscv,plic0` also, even if SiFive practically sets RISC-V standards.